packages/remote-config/lib/index.js

```diff
@@ -116,36 +184,73 @@ class FirebaseConfigModule extends FirebaseModule {
+    );
   }
 
-  setConfigSettings(settings = {}) {
-    if (!isObject(settings) || !hasOwnProperty(settings, 'isDeveloperModeEnabled')) {
-      throw new Error(
-        "firebase.remoteConfig().setConfigSettings(): 'settings' must be an object with a 'isDeveloperModeEnabled' key.",
-      );
+  /**
+   * Deletes all activated, fetched and defaults configs and resets all Firebase Remote Config settings.
+   * @returns {Promise<null>}
+   */
+  reset() {
+    if (isIOS) {
+      console.warn('firebase.remoteConfig().reset() is not yet supported on iOS.');
+      return Promise.resolve(null);
     }
 
-    if (!isBoolean(settings.isDeveloperModeEnabled)) {
-      throw new Error(
-        "firebase.remoteConfig().setConfigSettings(): 'settings.isDeveloperModeEnabled' must be a boolean value.",
+    return this._promiseWithConstants(this.native.reset());
+  }
+
+  setConfigSettings(settings) {
+    const nativeSettings = {};
+
+    if (!isObject(settings)) {
+      throw new Error('firebase.remoteConfig().settings: must set an object.');
+    }
+
+    if (hasOwnProperty(settings, 'isDeveloperModeEnabled')) {
+      console.warn(
+        "firebase.remoteConfig().setConfigSettings(): 'settings.isDeveloperModeEnabled' has now been removed. Please consider setting 'settings.minimumFetchIntervalMillis'",
       );
     }
 
-    if (
-      hasOwnProperty(settings, 'minimumFetchInterval') &&
-      !isNumber(settings.minimumFetchInterval)
-    ) {
-      throw new Error(
-        "firebase.remoteConfig().setConfigSettings(): 'settings.minimumFetchInterval' must be a number value.",
+    if (hasOwnProperty(settings, 'minimumFetchInterval')) {
+      console.warn(
+        "firebase.remoteConfig().setConfigSettings(): 'settings.minimumFetchInterval' has now been removed. Please consider setting 'settings.minimumFetchIntervalMillis'",
       );
     }
 
-    return this._promiseWithConstants(this.native.setConfigSettings(settings));
+    if (hasOwnProperty(settings, 'minimumFetchIntervalMillis')) {
+      if (!isNumber(settings.minimumFetchIntervalMillis)) {
+        throw new Error(
+          "firebase.remoteConfig().setConfigSettings(): 'settings.minimumFetchIntervalMillis' must be a number type in milliseconds.",
+        );
+      } else {
+        //iOS & Android expect seconds
+        nativeSettings.minimumFetchInterval = settings.minimumFetchIntervalMillis / 1000;
+      }
+    }
+
+    if (hasOwnProperty(settings, 'fetchTimeMillis')) {
+      if (!isNumber(settings.fetchTimeMillis)) {
+        throw new Error(
+          "firebase.remoteConfig().setConfigSettings(): 'settings.fetchTimeMillis' must be a number type in milliseconds.",
+        );
+      } else {
+        //iOS & Android expect seconds
+        nativeSettings.fetchTimeout = settings.fetchTimeMillis / 1000;
+      }
+    }
+
+    this._settings = settings;
```

Comment:
Should probably set this after the following promise completes successfully, so that it doesn't set it iffor some reason native errors

Reply:
Alternatively promiseWithConstants could pull the updated settings from native also and update the class property